logstash-core-event-java/src/main/java/com/logstash/ext/JrubyEventExtLibrary.java

```diff
@@ -140,6 +132,8 @@ public IRubyObject ruby_set_field(ThreadContext context, RubyString reference, I
                     this.event.setField(r, RubyToJavaConverter.convertToList((RubyArray) value));
                 } else if (value instanceof RubyHash) {
                     this.event.setField(r, RubyToJavaConverter.convertToMap((RubyHash) value));
+                } else if (value instanceof RubyBoolean) {
+                    this.event.setField(r, ((RubyBoolean) value).toJava(boolean.class));
                 } else if (value.isNil()) {
                     this.event.setField(r, null);
                 } else {
```

Comment:
Consider whether you need to provide for: RubySymbol, RubyBignum and RubyBigDecimal